### ISSUE TITLE
How to re-arrange oapi-codegen based petstore example using the Repository Pattern?

### ISSUE THREAD

**Issue by suntong** [ThreeDotsLabs/wild-workouts-go-ddd-example#70]
The [oapi-codegen based petstore example](https://github.com/deepmap/oapi-codegen/tree/master/examples/petstore-expanded) has a very simple and straightforward file structure. Using the fiber server as an example, as far as the server end is concerned, here are all the files it needs:

```
`--petstore-expanded.yaml
`--fiber
| `--petstore.go
| `--petstore_test.go
| `--api
| | `--petstore.go
| | `--petstore-server.gen.go
| | `--server.cfg.yaml
| | `--petstore-types.gen.go
| | `--types.cfg.yaml
```

How to re-arrange oapi-codegen based petstore example using the Repository Pattern?

I tried to split the code as per the [Repository Pattern](https://threedots.tech/post/repository-pattern-in-go/) myself following the folder structure in #69:

```
`--api
| `--openapi
| | `--petstore-expanded.yaml
`--internal
| `--petstore
| | `--main.go
| | `--main_test.go
| | `--core
| | | `--app
| | | `--domain
| | | | `--petstore.go
| | | `--services
| | `--interfaces
| | | `--adapters
| | | `--ports
| | | | `--petstore-server.gen.go
| | | | `--petstore-types.gen.go
| | | | `--server.cfg.yaml
| | | | `--types.cfg.yaml

```

However in file `domain/petstore.go` (changed a bit from [here](https://github.com/deepmap/oapi-codegen/blob/af55078a5e93787f522e8d26dd550ebf28d4deb1/examples/petstore-expanded/fiber/api/petstore.go#L44-L45)):

```go
// FindPets implements all the handlers in the ServerInterface
func (p *PetStore) FindPets(c *fiber.Ctx, params ports.FindPetsParams) error {
```

I.e., I have to use the type `FindPetsParams` defined in `ports` (`ports/petstore-types.gen.go`) in domain, and then extract the server agnostic part of algorithm into function `FindPets` in the domain package, and call such domain `FindPets` function in `http.go` file within the `ports` package.

In other words, I'll be doing import cycling, which is not allowed in Go. So, all in all,

How to re-arrange oapi-codegen based petstore example using the Repository Pattern? thx.

**Comment by m110** [comment]
Hey. 

First of all, your domain should have no relation to fiber, since it's a http-specific library. Keep fiber imports in ports and nowhere else. 

Then, you need to define the Params inside domain and map between it and fiber's Params in ports. Or you can have domain constructors that take arguments and just pass them from Params.

**Comment by suntong** [comment]
OK. thanks.
